studio/components/layouts/ProjectLayout/RestoringState.tsx

```diff
@@ -0,0 +1,89 @@
+import { FC, useEffect, useRef, useState } from 'react'
+import { Button, IconCheckCircle, IconLoader } from '@supabase/ui'
+
+import { useStore } from 'hooks'
+import { API_URL, PROJECT_STATUS } from 'lib/constants'
+import { getWithTimeout } from 'lib/common/fetch'
+
+interface Props {}
+
+const RestoringState: FC<Props> = ({}) => {
+  const { app, ui } = useStore()
+  const project = ui.selectedProject
+  const checkServerInterval = useRef<number>()
+
+  const [loading, setLoading] = useState(false)
+  const [isCompleted, setIsCompleted] = useState(false)
+
+  useEffect(() => {
+    checkServerInterval.current = window.setInterval(checkServer, 4000)
+    return () => clearInterval(checkServerInterval.current)
+  }, [])
+
+  async function checkServer() {
+    if (!project) return
+
+    const projectStatus = await getWithTimeout(`${API_URL}/projects/${project.ref}/status`, {
+      timeout: 2000,
+    })
+    if (projectStatus && !projectStatus.error) {
+      const { status } = projectStatus
+      if (status === PROJECT_STATUS.ACTIVE_HEALTHY) {
+        clearInterval(checkServerInterval.current)
+        setIsCompleted(true)
+      }
+    }
+  }
+
+  const onConfirm = async () => {
+    setLoading(true)
+    await app.projects.fetchDetail(project?.ref ?? '')
+  }
+
+  return (
+    <div className="flex items-center justify-center h-full">
+      <div
+        className={[
+          'grid grid-cols-12 gap-4',
+          'bg-scale-300 border border-scale-400 rounded-md w-3/4 lg:w-1/2 px-8 py-6',
+        ].join(' ')}
+      >
+        {isCompleted ? (
+          <>
+            <div className="col-span-1">
+              <IconCheckCircle className="text-brand-900" strokeWidth={2} />
+            </div>
+            <div className="col-span-11 space-y-1">
+              <p>Restoration complete!</p>
+              <p className="text-sm text-scale-1100">
+                Your project has been successfully restored and is now back online.
+              </p>
+            </div>
+          </>
+        ) : (
+          <>
+            <div className="col-span-1">
+              <IconLoader className="animate-spin" />
+            </div>
+            <div className="col-span-11 space-y-1">
+              <p>Restoration in progress</p>
+              <p className="text-sm text-scale-1100">
+                Restoration can take from a few minutes up to several hours depending on the size of
+                your database. Your project will be offline while the restoration is running.
+              </p>
+            </div>
+          </>
+        )}
+        {isCompleted && (
+          <div className="col-start-2 col-span-11">
+            <Button disabled={loading} loading={loading} onClick={onConfirm}>
+              Confirm
```

Comment:
'Confirm' seems a little strange – maybe 'Okay' or 'Return Home' or similar :)

Reply:
> The icon spacing feels a little large on the left compared to the right:

Hmm yeah - this i think i'll leave it first though since its a UI component and may affect other parts of the dashboard, but i'll make the field slightly longer too for now
<img width="435" alt="image" src="https://user-images.githubusercontent.com/19742402/192722656-af02f4f3-1ee3-434f-9185-1c72de1dc4d7.png">

> Is this state needed? Is there a downside I'm not thinking of to just selecting the current date as the default?

It's mainly to make the date selection a little more explicit and make the PITR selection flow more step-wise - makes it slightly less prone to errors. Chances are the user won't be selecting the current date to restore from too.

'Confirm' seems a little strange – maybe 'Okay' or 'Return Home' or similar :)

Ahh yeah that's fair will update to 'Return to project'